src/events.rs

```diff
@@ -239,11 +239,28 @@ impl IAudioSessionEvents_Impl for AudioSessionEvents {
         trace!("New channel volume for channel: {}", changedchannel);
         let volslice =
             unsafe { slice::from_raw_parts(newchannelvolumearray, channelcount as usize) };
-        let newvol = volslice[changedchannel as usize];
-        if let Some(callbacks) = &mut self.callbacks.upgrade() {
-            if let Some(callback) = &callbacks.channel_volume {
-                let context = unsafe { *eventcontext };
-                callback(changedchannel as usize, newvol, context);
+        let changed_indices = if (changedchannel as usize) < volslice.len() {
+            // this specific channel
+            vec![changedchannel as usize]
```

Comment:
It would be nicer if we don't need to allocate a vector here in the callback. I would move this check inside the block further down, something like:
```
if let Some(callbacks) = &mut self.callbacks.upgrade() {
                if let Some(callback) = &callbacks.channel_volume {
                    let context = unsafe { *eventcontext };
                    if changedchannel == u32::MAX {
                      // loop that calls callback for each channel
                    }
                    else if (changedchannel as usize) < volslice.len() {
                      callback(changedchannel as usize, newvol, context);
                    }
                    else { warn!("oh no!") }
                }
            }
```

Reply:
Thanks, the commit is updated accordingly.